test/lit/passes/optimize-instructions-ignore-traps.wast

```diff
@@ -759,4 +754,65 @@
     (i32.const 0)
   )
  )
+
+ ;; CHECK:      (func $optimize-bulk-memory-fill (param $dst i32) (param $val i32) (param $sz i32)
+ ;; CHECK-NEXT:  (block
+ ;; CHECK-NEXT:   (drop
+ ;; CHECK-NEXT:    (local.get $dst)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:   (drop
+ ;; CHECK-NEXT:    (i32.const 0)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (block
+ ;; CHECK-NEXT:   (drop
+ ;; CHECK-NEXT:    (local.get $dst)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:   (drop
+ ;; CHECK-NEXT:    (local.get $val)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (block
+ ;; CHECK-NEXT:   (drop
+ ;; CHECK-NEXT:    (i32.load
+ ;; CHECK-NEXT:     (i32.const -1)
+ ;; CHECK-NEXT:    )
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:   (drop
+ ;; CHECK-NEXT:    (i32.const 0)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT:  (memory.fill
+ ;; CHECK-NEXT:   (i32.const 0)
+ ;; CHECK-NEXT:   (i32.const 0)
+ ;; CHECK-NEXT:   (i32.load
+ ;; CHECK-NEXT:    (i32.const -1)
+ ;; CHECK-NEXT:   )
+ ;; CHECK-NEXT:  )
+ ;; CHECK-NEXT: )
+ (func $optimize-bulk-memory-fill (param $dst i32) (param $val i32) (param $sz i32)
+  (memory.fill ;; drops
+    (local.get $dst)
+    (i32.const 0)
+    (i32.const 0)
+  )
+
+  (memory.fill ;; drops
+    (local.get $dst)
+    (local.get $val)
+    (i32.const 0)
+  )
+
+  (memory.fill ;; drops
+    (i32.load (i32.const -1))
+    (i32.const 0)
+    (i32.const 0)
+  )
+
+  (memory.fill ;; skip
+    (i32.const 0)
+    (i32.const 0)
+    (i32.load (i32.const -1))
```

Comment:
Could this be a `local.get $sz`? I notice we haven't used the $sz parameter, and this load from a negative index looks like it should be testing something about the load, but actually the point is just that it's not a constant, right?

Reply:
Yeah, makes sense